/arch/TriCore/TriCoreMapping.o: In function `getVectorMinNumElements': TriCoreMapping.c:(.text+0x440): multiple definition of `getVectorMinNumElements'

From https://github.com/capstone-engine/capstone/actions/runs/5597666737/jobs/10236327487

```
In file included from arch/TriCore/TriCoreMapping.c:12:0:
arch/TriCore/../../cs_simple_types.h:325:13: warning: no previous prototype for ‘isValid’ [-Wmissing-prototypes]
 inline bool isValid(cs_data_type SimpleTy)
             ^
arch/TriCore/../../cs_simple_types.h:332:13: warning: no previous prototype for ‘isFloatingPoint’ [-Wmissing-prototypes]
 inline bool isFloatingPoint(cs_data_type SimpleTy)
             ^
arch/TriCore/../../cs_simple_types.h:343:13: warning: no previous prototype for ‘isInteger’ [-Wmissing-prototypes]
 inline bool isInteger(cs_data_type SimpleTy)
             ^
arch/TriCore/../../cs_simple_types.h:358:13: warning: no previous prototype for ‘isScalarInteger’ [-Wmissing-prototypes]
 inline bool isScalarInteger(cs_data_type SimpleTy)
             ^
arch/TriCore/../../cs_simple_types.h:365:13: warning: no previous prototype for ‘isVector’ [-Wmissing-prototypes]
 inline bool isVector(cs_data_type SimpleTy)
             ^
arch/TriCore/../../cs_simple_types.h:373:13: warning: no previous prototype for ‘isScalableVector’ [-Wmissing-prototypes]
 inline bool isScalableVector(cs_data_type SimpleTy)
             ^
arch/TriCore/../../cs_simple_types.h:379:13: warning: no previous prototype for ‘isFixedLengthVector’ [-Wmissing-prototypes]
 inline bool isFixedLengthVector(cs_data_type SimpleTy)
             ^
arch/TriCore/../../cs_simple_types.h:386:13: warning: no previous prototype for ‘is16BitVector’ [-Wmissing-prototypes]
 inline bool is16BitVector(cs_data_type SimpleTy)
             ^
arch/TriCore/../../cs_simple_types.h:395:13: warning: no previous prototype for ‘is32BitVector’ [-Wmissing-prototypes]
 inline bool is32BitVector(cs_data_type SimpleTy)
             ^
arch/TriCore/../../cs_simple_types.h:407:13: warning: no previous prototype for ‘is64BitVector’ [-Wmissing-prototypes]
 inline bool is64BitVector(cs_data_type SimpleTy)
             ^
arch/TriCore/../../cs_simple_types.h:421:13: warning: no previous prototype for ‘is128BitVector’ [-Wmissing-prototypes]
 inline bool is128BitVector(cs_data_type SimpleTy)
             ^
arch/TriCore/../../cs_simple_types.h:436:13: warning: no previous prototype for ‘is256BitVector’ [-Wmissing-prototypes]
 inline bool is256BitVector(cs_data_type SimpleTy)
             ^
arch/TriCore/../../cs_simple_types.h:452:13: warning: no previous prototype for ‘is512BitVector’ [-Wmissing-prototypes]
 inline bool is512BitVector(cs_data_type SimpleTy)
             ^
arch/TriCore/../../cs_simple_types.h:468:13: warning: no previous prototype for ‘is1024BitVector’ [-Wmissing-prototypes]
 inline bool is1024BitVector(cs_data_type SimpleTy)
             ^
arch/TriCore/../../cs_simple_types.h:482:13: warning: no previous prototype for ‘is2048BitVector’ [-Wmissing-prototypes]
 inline bool is2048BitVector(cs_data_type SimpleTy)
             ^
arch/TriCore/../../cs_simple_types.h:495:21: warning: no previous prototype for ‘getVectorElementType’ [-Wmissing-prototypes]
 inline cs_data_type getVectorElementType(cs_data_type SimpleTy)
                     ^
arch/TriCore/../../cs_simple_types.h:687:17: warning: no previous prototype for ‘getVectorMinNumElements’ [-Wmissing-prototypes]
 inline unsigned getVectorMinNumElements(cs_data_type SimpleTy)
                 ^
  GEN     capstone.pc
  LINK    libcapstone.so.5
  AR      libcapstone.a
/opt/rh/devtoolset-2/root/usr/bin/ar: creating ./libcapstone.a
./arch/TriCore/TriCoreMapping.o: In function `isValid':
TriCoreMapping.c:(.text+0x0): multiple definition of `isValid'
./arch/ARM/ARMMapping.o:ARMMapping.c:(.text+0x240): first defined here
./arch/TriCore/TriCoreMapping.o: In function `isFloatingPoint':
TriCoreMapping.c:(.text+0x10): multiple definition of `isFloatingPoint'
./arch/ARM/ARMMapping.o:ARMMapping.c:(.text+0x250): first defined here
./arch/TriCore/TriCoreMapping.o: In function `isInteger':
TriCoreMapping.c:(.text+0x40): multiple definition of `isInteger'
./arch/ARM/ARMMapping.o:ARMMapping.c:(.text+0x280): first defined here
./arch/TriCore/TriCoreMapping.o: In function `isScalarInteger':
TriCoreMapping.c:(.text+0x70): multiple definition of `isScalarInteger'
./arch/ARM/ARMMapping.o:ARMMapping.c:(.text+0x2b0): first defined here
./arch/TriCore/TriCoreMapping.o: In function `isVector':
TriCoreMapping.c:(.text+0x80): multiple definition of `isVector'
./arch/ARM/ARMMapping.o:ARMMapping.c:(.text+0x2c0): first defined here
./arch/TriCore/TriCoreMapping.o: In function `isScalableVector':
TriCoreMapping.c:(.text+0x90): multiple definition of `isScalableVector'
./arch/ARM/ARMMapping.o:ARMMapping.c:(.text+0x2d0): first defined here
./arch/TriCore/TriCoreMapping.o: In function `isFixedLengthVector':
TriCoreMapping.c:(.text+0xa0): multiple definition of `isFixedLengthVector'
./arch/ARM/ARMMapping.o:ARMMapping.c:(.text+0x2e0): first defined here
./arch/TriCore/TriCoreMapping.o: In function `is16BitVector':
TriCoreMapping.c:(.text+0xb0): multiple definition of `is16BitVector'
./arch/ARM/ARMMapping.o:ARMMapping.c:(.text+0x2f0): first defined here
./arch/TriCore/TriCoreMapping.o: In function `is32BitVector':
TriCoreMapping.c:(.text+0xe0): multiple definition of `is32BitVector'
./arch/ARM/ARMMapping.o:ARMMapping.c:(.text+0x[320](https://github.com/capstone-engine/capstone/actions/runs/5597666737/jobs/10236327487#step:7:321)): first defined here
./arch/TriCore/TriCoreMapping.o: In function `is64BitVector':
TriCoreMapping.c:(.text+0x120): multiple definition of `is64BitVector'
./arch/ARM/ARMMapping.o:ARMMapping.c:(.text+0x[360](https://github.com/capstone-engine/capstone/actions/runs/5597666737/jobs/10236327487#step:7:361)): first defined here
./arch/TriCore/TriCoreMapping.o: In function `is128BitVector':
TriCoreMapping.c:(.text+0x180): multiple definition of `is128BitVector'
./arch/ARM/ARMMapping.o:ARMMapping.c:(.text+0x3c0): first defined here
./arch/TriCore/TriCoreMapping.o: In function `is256BitVector':
TriCoreMapping.c:(.text+0x1e0): multiple definition of `is256BitVector'
./arch/ARM/ARMMapping.o:ARMMapping.c:(.text+0x420): first defined here
./arch/TriCore/TriCoreMapping.o: In function `is512BitVector':
TriCoreMapping.c:(.text+0x250): multiple definition of `is512BitVector'
./arch/ARM/ARMMapping.o:ARMMapping.c:(.text+0x490): first defined here
./arch/TriCore/TriCoreMapping.o: In function `is1024BitVector':
TriCoreMapping.c:(.text+0x2c0): multiple definition of `is1024BitVector'
./arch/ARM/ARMMapping.o:ARMMapping.c:(.text+0x500): first defined here
./arch/TriCore/TriCoreMapping.o: In function `is2048BitVector':
TriCoreMapping.c:(.text+0x320): multiple definition of `is2048BitVector'
./arch/ARM/ARMMapping.o:ARMMapping.c:(.text+0x560): first defined here
./arch/TriCore/TriCoreMapping.o: In function `getVectorElementType':
TriCoreMapping.c:(.text+0x[380](https://github.com/capstone-engine/capstone/actions/runs/5597666737/jobs/10236327487#step:7:381)): multiple definition of `getVectorElementType'
./arch/ARM/ARMMapping.o:ARMMapping.c:(.text+0x5c0): first defined here
./arch/TriCore/TriCoreMapping.o: In function `getVectorMinNumElements':
TriCoreMapping.c:(.text+0x440): multiple definition of `getVectorMinNumElements'
./arch/ARM/ARMMapping.o:ARMMapping.c:(.text+0x680): first defined here
collect2: error: ld returned 1 exit status
make: *** [libcapstone.so.5] Error 1
/opt/python/cp36-cp36m/lib/python3.6/site-packages/setuptools/dist.py:484: UserWarning: Normalizing '5.0.0post1' to '5.0.0.post1'
  warnings.warn(tmpl.format(**locals()))
error: [Errno 2] No such file or directory: 'libcapstone.so.5'
Error: Process completed with exit code 1.
0s
0s
```
cc @Rot127

--- a/.github/workflows/python-publish.yml
+++ b/.github/workflows/python-publish.yml
@@ -16,6 +16,8 @@ jobs:
 
     - name: Set up Python
       uses: actions/setup-python@v4
+      # Python is included in manylinux
+      if: matrix.os != 'ubuntu-latest'
       with:
         python-version: '3.x'
 
@@ -31,19 +33,21 @@ jobs:
 
     - name: Install dependencies
       run: |
-        pip install setuptools wheel
+        if [ ${{ matrix.os }} != 'ubuntu-latest' ]; then
+          pip install setuptools wheel
+        fi
     - name: Build distribution 📦
       shell: bash 
       run: |
         if [ ${{ matrix.platform }} == 'x32' ] && [ ${{ matrix.os }} == 'windows-latest' ]; then
              cd bindings/python && python setup.py build -p win32 bdist_wheel -p win32
         elif [ ${{ matrix.platform }} == 'x32' ] && [ ${{ matrix.os }} == 'ubuntu-latest' ]; then
-             docker run --rm -v `pwd`/:/work dockcross/manylinux1-x86 > ./dockcross
+             docker run --rm -v `pwd`/:/work dockcross/manylinux2014-x86 > ./dockcross
              chmod +x ./dockcross
              chmod +x bindings/python/build_wheel.sh
              ./dockcross bindings/python/build_wheel.sh
         elif [ ${{ matrix.platform }} == 'x64' ] && [ ${{ matrix.os }} == 'ubuntu-latest' ]; then
-             docker run --rm -v `pwd`/:/work dockcross/manylinux1-x64 > ./dockcross
+             docker run --rm -v `pwd`/:/work dockcross/manylinux_2_28-x64 > ./dockcross
              chmod +x ./dockcross
              chmod +x bindings/python/build_wheel.sh
              ./dockcross bindings/python/build_wheel.sh

--- a/bindings/python/build_wheel.sh
+++ b/bindings/python/build_wheel.sh
@@ -2,14 +2,15 @@
 set -e -x
 
 cd bindings/python
-sudo rm /usr/bin/python && sudo ln -s /opt/python/cp27-cp27m/bin/python /usr/bin/python; python -V
-
-# Compile wheels
-if [ -f /opt/python/cp36-cp36m/bin/python ];then
-  /opt/python/cp36-cp36m/bin/python setup.py bdist_wheel
+if [ -f /opt/python/cp311-cp311/bin/python3 ];then
+  # Use manylinux Python
+  /opt/python/cp311-cp311/bin/python3 -m pip install wheel
+  /opt/python/cp311-cp311/bin/python3 setup.py bdist_wheel
 else
+  python3 -m pip install wheel
   python3 setup.py bdist_wheel
 fi
+
 cd dist
 auditwheel repair *.whl
 mv -f wheelhouse/*.whl .

--- a/bindings/python/setup.py
+++ b/bindings/python/setup.py
@@ -206,7 +206,14 @@ if 'bdist_wheel' in sys.argv and '--plat-name' not in sys.argv:
         # linux builds should be built in the centos 5 vm for maximum compatibility
         # see https://github.com/pypa/manylinux
         # see also https://github.com/angr/angr-dev/blob/master/bdist.sh
-        sys.argv.insert(idx + 1, 'manylinux1_' + platform.machine())
+        machine = platform.machine()
+        if machine == "i686":
+            # manylinux2014 is the last which supports x86 in dockercross
+            sys.argv.insert(idx + 1, 'manylinux2014-x86')
+        elif machine == "x86_64":
+            sys.argv.insert(idx + 1, 'manylinux_2_28-x64')
+        else:
+            raise ValueError("Unknown machine: " + machine)
     else:
         # https://www.python.org/dev/peps/pep-0425/
         sys.argv.insert(idx + 1, name.replace('.', '_').replace('-', '_'))

